### PR TITLE
Refactor backend

### PR DESCRIPTION
**Pull request by DenisaCG**
Refactor backend functionalities to use `fsspec`. The `obstore` package was kept for certain functionalities as it has integrated pagination, so it is still used to list the contents given a path, retrieve contents of a file and get the presigned link of an object, while the `fsspec` package is used for creating, saving, renaming, copying and deleting.

As a workaround for dealing with directories, they are created as objects having a key which ends with the suffix `/.jupyter-drives-fixDir`. This way they will be hidden inside the `DrivesFileBrowser`, but users can still create, rename, copy and delete directorie.

### CODE REVIEW

File: jupyter_drives/manager.py [hunk 1]
```diff
@@ -41,12 +43,17 @@ def __init__(self, config: traitlets.config.Config) -> None:
...
         self._content_managers = {}
         self._max_files_listed = 1000
+        
+        # instate fsspec file system
+        self._file_system = fsspec.filesystem(self._config.provider, asynchronous=True)
 
         # initiate aiobotocore session if we are dealing with S3 drives
         if self._config.provider == 's3':
             if self._config.access_key_id and self._config.secret_access_key:
+                self._fixDir_suffix = '/.jupyter-drives-fixDir' # fix for creating directories
```
Comment (afshin) [inline jupyter_drives/manager.py:53]: Could we make this a constant in this file (e.g., `EMPTY_DIR = '/.jupyter_drives_fix_dir'` or something like this?

File: jupyter_drives/manager.py [hunk 2]
```diff
@@ -216,56 +223,47 @@ async def get_contents(self, drive_name, path):
...
-                    remaining_files = self._max_files_listed - no_batches*chunk_size
-                    
-                # if content exists we are dealing with a directory
-                if isDir is False and batch: 
-                    isDir = True
-                    emptyDir = False
+            is_dir = await self._file_system._isdir(drive_name + '/' + path)
+
+            if is_dir == True:
+                chunk_size = 100
```
Comment (afshin) [inline jupyter_drives/manager.py:229]:
```suggestion
                chunk_size = 1024
```

Comment (afshin) [reply jupyter_drives/manager.py:229]: I feel like 1000 is still doable in the file browser ... do you think its performance would be terrible?

I don't have a strong conviction, it's just that 100 seems small to me.

**Review by afshin — APPROVED** [review]
Thank you!

File: jupyter_drives/manager.py (hunk 1 above)
Comment (DenisaCG) [reply jupyter_drives/manager.py]: Thanks for the suggestion. I pushed a commit to switch to the use of the `EMPTY_DIR_SUFFIX = '/.jupyter_drives_fix_dir'` constant.

**Comment by DenisaCG** [comment]
Thank you for the review!

**Pull request merged**